src/FlatParse/Basic.hs

```diff
@@ -200,71 +208,85 @@ import FlatParse.Internal.UnboxedNumerics
 --------------------------------------------------------------------------------
 
```

Comment:
We introduce a separate type alias for the outer state token passing to make it easier to just thread the state token if you dont care about the actual result type.